In [15]:
import pandas as pd

path = '/home/tito/repositorios/Curso-NumPy-Pandas-Matplotlib/Online_Retail.csv'
df = pd.read_csv(path, encoding='latin1')

# Esto es una manipulación, puedo crear una columna nueva a partir de otras
df['TotalPrice'] = df['Quantity'] * df['UnitPrice']
df.head()

,InvoiceNo,Unnamed: 1,DescriptiStockCodeon,Quantity,InvoiceDate,UnitPrice,CustomerID,Country,TotalPrice
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,12/1/10 8:26,2.55,17850.0,United Kingdom,15.30
1,536365,71053,WHITE METAL LANTERN,6,12/1/10 8:26,3.39,17850.0,United Kingdom,20.34
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,12/1/10 8:26,2.75,17850.0,United Kingdom,22.00
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,12/1/10 8:26,3.39,17850.0,United Kingdom,20.34
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,12/1/10 8:26,3.39,17850.0,United Kingdom,20.34


In [16]:
df['HighValue'] = df['TotalPrice'] > 100

# La columna HighValue trae True si TotalPrice es > 100
df['HighValue'].value_counts()

HighValue
False    528355
True      13554
Name: count, dtype: int64

In [17]:
# Con esto traigo información sobre el df
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 10 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   InvoiceNo             541909 non-null  str    
 1   Unnamed: 1            541909 non-null  str    
 2   DescriptiStockCodeon  540455 non-null  str    
 3   Quantity              541909 non-null  int64  
 4   InvoiceDate           541909 non-null  str    
 5   UnitPrice             541909 non-null  float64
 6   CustomerID            406829 non-null  float64
 7   Country               541909 non-null  str    
 8   TotalPrice            541909 non-null  float64
 9   HighValue             541909 non-null  bool   
dtypes: bool(1), float64(3), int64(1), str(5)
memory usage: 37.7 MB


Vemos que:

4 InvoiceDate 541909 non-ull float64

esto es una fecha, vamos a darle formato de fecha

In [18]:
df['InvoiceDate'] = pd.to_datetime(df['InvoiceDate'])
df.info()

/tmp/ipykernel_11868/40610843.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['InvoiceDate'] = pd.to_datetime(df['InvoiceDate'])


<class 'pandas.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 10 columns):
 #   Column                Non-Null Count   Dtype         
---  ------                --------------   -----         
 0   InvoiceNo             541909 non-null  str           
 1   Unnamed: 1            541909 non-null  str           
 2   DescriptiStockCodeon  540455 non-null  str           
 3   Quantity              541909 non-null  int64         
 4   InvoiceDate           541909 non-null  datetime64[us]
 5   UnitPrice             541909 non-null  float64       
 6   CustomerID            406829 non-null  float64       
 7   Country               541909 non-null  str           
 8   TotalPrice            541909 non-null  float64       
 9   HighValue             541909 non-null  bool          
dtypes: bool(1), datetime64[us](1), float64(3), int64(1), str(4)
memory usage: 37.7 MB


In [14]:
df

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country,TotalPrice,HighValue
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,2010-12-01 08:26:00,2.55,17850.0,United Kingdom,15.30,False
1,536365,71053,WHITE METAL LANTERN,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom,20.34,False
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,2010-12-01 08:26:00,2.75,17850.0,United Kingdom,22.00,False
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom,20.34,False
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom,20.34,False
...,...,...,...,...,...,...,...,...,...,...
541904,581587,22613,PACK OF 20 SPACEBOY NAPKINS,12,2011-12-09 12:50:00,0.85,12680.0,France,10.20,False
541905,581587,22899,CHILDREN'S APRON DOLLY GIRL,6,2011-12-09 12:50:00,2.10,12680.0,France,12.60,False
541906,581587,23254,CHILDRENS CUTLERY DOLLY GIRL,4,2011-12-09 12:50:00,4.15,12680.0,France,16.60,False
541907,581587,23255,CHILDRENS CUTLERY CIRCUS PARADE,4,2011-12-09 12:50:00,4.15,12680.0,France,16.60,False
